### PR TITLE
HeppyCore updates

### PR DESCRIPTION
**Pull request by GaelTouquet**

### CODE REVIEW

File: PhysicsTools/HeppyCore/python/framework/event.py [hunk 1]
```diff
@@ -1,47 +1,124 @@
...
+        '''returns a dict of printable information of an event
+        arguments
+        * subname is used when called recursively and is the name of the parent object'''
+        selected_attrs = copy.copy(self.__dict__) #initial selection of what we can print
+        selected_attrs.pop('setup') #get rid of some bits
+        selected_attrs.pop('input')
+        
+        # Colin: the following are unused: 
+        matched_attrs = dict() #this applies pattern matching to obtain a subset of selected_attrs
+        print_attrs=dict() #ready for printing
```
Comment (gpetruc) [inline PhysicsTools/HeppyCore/python/framework/event.py:63]: if they're unused then why add them?

File: PhysicsTools/HeppyCore/python/framework/eventsfwlite.py [hunk 2]
```diff
@@ -15,16 +18,19 @@
...
-	if options is not None :
-		if not hasattr(options,"inputFiles"):
-		 	options.inputFiles=files
-		if not hasattr(options,"maxEvents"):
-			options.maxEvents = 0	
-		if not hasattr(options,"secondaryInputFiles"):
-			options.secondaryInputFiles = []
-	        self.events = FWLiteEvents(options=options)
+        logging.info('opening input files:')
+        logging.info(pprint.pformat(files))
```
Comment (gpetruc) [inline PhysicsTools/HeppyCore/python/framework/eventsfwlite.py:22]: shouldn't they be sent in just one message, e.g. `'opening input files: %s' % pprint.pformat(files)` ?

File: PhysicsTools/HeppyCore/python/framework/eventsfwlite.py [hunk 3]
```diff
@@ -15,16 +18,19 @@
...
+        if options is not None :
+            if not hasattr(options,"inputFiles"):
+                options.inputFiles=files
+            if not hasattr(options,"maxEvents"):
+                options.maxEvents = 0	
+            if not hasattr(options,"secondaryInputFiles"):
+                options.secondaryInputFiles = []
```
Comment (gpetruc) [inline PhysicsTools/HeppyCore/python/framework/eventsfwlite.py:29]: if you add logging, I'd suggest adding
````
elif options.secondaryInputFiles: # only if it's a non-empty list
    logging.info('using secondary input files: %s' % pprint.pformat(options.secondaryInputFiles))
````

File: PhysicsTools/HeppyCore/python/framework/looper.py [hunk 4]
```diff
@@ -195,7 +203,9 @@ def loop(self):
                     else:
                         print 'event %d (%.1f ev/s)' % (iEv, (iEv-self.start_time_event)/float(timeit.default_timer() - self.start_time))
 
-                self.process( iEv )
+                if at_firstEvent:
+                    self.logger.info('done')
```
Comment (gpetruc) [inline PhysicsTools/HeppyCore/python/framework/looper.py:207]: maybe `done first event` ?

**Review by gpetruc — COMMENTED** [review]
I'm suggesting a few small improvements - but they can also come in a second PR if you prefer.

File: PhysicsTools/HeppyCore/python/framework/event.py (hunk 1 above)
Comment (cbernet) [reply PhysicsTools/HeppyCore/python/framework/event.py:63]: hehe indeed. I guess this is a remnant of the prototyping of the code. removed.

File: PhysicsTools/HeppyCore/python/framework/eventsfwlite.py (hunk 2 above)
Comment (cbernet) [reply PhysicsTools/HeppyCore/python/framework/eventsfwlite.py:22]: done

File: PhysicsTools/HeppyCore/python/framework/eventsfwlite.py (hunk 3 above)
Comment (cbernet) [reply PhysicsTools/HeppyCore/python/framework/eventsfwlite.py:29]: Done, but cannot really test it (never used this option)

File: PhysicsTools/HeppyCore/python/framework/looper.py (hunk 4 above)
Comment (cbernet) [reply PhysicsTools/HeppyCore/python/framework/looper.py:207]: done

**Comment by cbernet** [comment]
can be closed, overridden by PR #740

**Pull request closed without merging**